sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -817,6 +892,27 @@ public void setValueDeserializer(String valueDeserializer) {
       return new TypedWithoutMetadata<>(this);
     }
 
+    private Schema fetchAvroSchema(String schemaRegistryUrl, String subject) {
+      SchemaRegistryClient registryClient;
```

Comment:
Can you refactor the code of `KafkaUnboundedReader#getSchemaRegistryClient` into a method that you use here to avoid the repeated code to get a client.